source/common/ssl/ssl_socket.cc

```diff
@@ -290,6 +306,30 @@ std::string SslSocket::getUriSanFromCertificate(X509* cert) {
   return "";
 }
 
+std::vector<std::string> SslSocket::getDnsSansFromCertificate(X509* cert) {
+  STACK_OF(GENERAL_NAME)* altnames = static_cast<STACK_OF(GENERAL_NAME)*>(
```

Comment:
Erm, you still didn't switch to ranged for loop (see https://github.com/envoyproxy/envoy/pull/3081).

Reply:
Isn't line 317 the ranged for loop you asked for?

Reply:
That's just a part of it. Please see how `SslSocket::getUriSanFromCertificate()` was rewritten earlier today with `bssl::UniquePtr<GENERAL_NAMES>()`.

Reply:
Updated to match getUriSanFromCertificate